test/exchange-files.js

```diff
@@ -94,13 +101,15 @@ const timeout = isCi ? 15 * min : 10 * min
 describe('exchange files', function () {
   this.timeout(timeout)
 
+  /** @type {Record<string, ('go' | 'js')[]>} */
```

Comment:
This seems to be an incorrect type for what you're trying to accomplish. lmk if you want to allow a 3-tuple.

```ts

type arr = ('go' | 'js')[]

const array: arr = ['go', 'js', 'go'] // False Positive
const array1: arr = ['go', 'go'] // True positive
const array2: arr = ['go', 'js'] // True positive
const array3: arr = ['js', 'go'] // True positive
const array4: arr = ['js', 'js'] // True positive

type goOrJs = 'go' | 'js'
type arrFix = [goOrJs, goOrJs]

const arrayFixed: arrFix = ['go', 'js', 'go'] // True Negative
const arrayFixedException1: arrFix = ['go', 'go'] // True positive
const arrayFixedException2: arrFix = ['go', 'js'] // True positive
const arrayFixedException3: arrFix = ['js', 'go'] // True positive
const arrayFixedException4: arrFix = ['js', 'js'] // True positive

```


**Playground Link:** [Provided](https://www.typescriptlang.org/play?#code/FAFwngDgpgBAhgJwTAvDAFAcgOYHtMwA+MmAVgM6YCUA2gLrDADGuAduSPEnGAFxfI0NHPgA0JCpnEjMdGAHp5MAGJwANuVgAFXOQCWIPQDcozNhwE8AjP0SCYwvFJJO5imABUEAV1gRdBsamLOycdjwATLZIqA4y0pJuSl6+MP76hiZmoZZgAMzR9sKS0q4KyT5+AZnB5mHcYAAshbHFlAmUSZ6VadVBjODQMHgA8ggAUuSxMkQSlKCQsHbKegAeraMT5OKbkwzZFuFgK6tQACaFJ63xc84yXSmwAHJQ2HA1B-UIPCfnAKKrJhQCCGNg2ARXIQ3e7lbqpdKBLIhQ4NX5nAFAkF6NhRCFra5ODqyWGPXoZfrIr4-Nb-QHA0GsAp49ZQkoufAPHoIj6U3JojH07GsZrM65ssidElcvpZIA)